core/drand_beacon.go

```diff
@@ -434,6 +442,92 @@ func (bp *BeaconProcess) newMetadata() *common.Metadata {
 	return metadata
 }
 
+func (bp *BeaconProcess) storePreviousFromNetwork(store chain.Store) error {
+	ctx := context.Background()
+
+	clkNow := bp.opts.clock.Now().Unix()
+	nextRound, nextRoundTime := chain.NextRound(clkNow, bp.group.Period, bp.group.GenesisTime)
```

Comment:
could we get `CurrentRound` instead and get one before that? That will _definitely_ have been created at this point and we don't have to worry about any clock shenanigans.

Reply:
`chain.CurrentRound` does kinda what the current code was doing when using `NextRound-1`.
However, it's more clear what the intent is. I switched to that.